openedx/core/djangoapps/user_api/views.py

```diff
@@ -249,10 +253,10 @@ def post(self, request):
 
         You must send all required form fields with the request.
 
-        You can optionally send an `analytics` param with a JSON-encoded
-        object with additional info to include in the registration analytics event.
-        Currently, the only supported field is "enroll_course_id" to indicate
-        that the user registered while enrolling in a particular course.
+        You can optionally send a "course_id" param to indicate in analytics
+        events that the user registered while enrolling in a particular course.
+        This can also be provided as a JSON-encoded "analytics" parameter with
+        an "enroll_course_id" field, but that approach is deprecated.
```

Comment:
The only client that uses the "analytics" parameter right now is the combined login/registration page.  It would be relatively cheap to change that call and remove the old parameter entirely.  If that's out of scope for this PR, could you create a dev story in our backlog to do this?